.github/workflows/shellcheck.yaml

```diff
@@ -0,0 +1,18 @@
+name: ShellCheck
+
+on: [push]
+
+permissions:
+ contents: read
+jobs:
+  shellcheck:
+    name: Run shellcheck
+    runs-on: ubuntu-latest
+    env:
+      GOPRIVATE: github.com/sagaxyz/*
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
+        env:
+          SHELLCHECK_OPTS: -e SC2086
```

Comment:
why not have the double quotes linter?

Reply:
I don't think we have any scripts in this repo anyway :)